scripts/update_readme_for_theme.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from argparse import ArgumentParser
+from importlib import import_module
+from inspect import getdoc
+from pathlib import Path
+
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from pygments.styles import get_style_by_name
+
+from a11y_pygments.utils.utils import find_all_themes_packages
+from a11y_pygments.utils.wcag_contrast import (
+    contrast_ratio,
+    get_wcag_level_large_text,
+    get_wcag_level_normal_text,
+    hex_to_rgb,
+    hexstr_without_hash,
+)
+
+# This script generates READMEs for themes from their style.py definitions
+
+HERE = Path(__file__).parent
+REPO = HERE.parent
+
+env = Environment(
+    loader=FileSystemLoader(HERE / "templates"), autoescape=select_autoescape()
+)
+
+
+def update_readme(theme):
+    """Given a theme module name, update that theme's README file on disk"""
+
+    # Get the theme colors
+    theme_style_module = import_module(f"a11y_pygments.{theme}.style")
+    color_cls = theme_style_module.Colors  # access to foreground colors
+    theme_cls = theme_style_module.Theme  # access to docstring
+    theme_kebab_case = theme.replace("_", "-")
+    style = get_style_by_name(
+        theme_kebab_case
+    )  # access to background and highlight colors
+
+    # Calculate contrast ratios and WCAG ratings for all foreground colors
+    foreground_colors = {}
+    for key in vars(color_cls):
+        value = getattr(color_cls, key)
+        if not callable(value) and not key.startswith("__"):
+            if value in foreground_colors:
+                continue
+            contrast = contrast_ratio(
+                hex_to_rgb(value), hex_to_rgb(style.background_color)
+            )
+            foreground_colors[value] = {
+                "hex": hexstr_without_hash(value),
+                "contrast_ratio": round(contrast, 1),
+                "wcag_level_normal_text": get_wcag_level_normal_text(contrast),
+                "wcag_level_large_text": get_wcag_level_large_text(contrast),
+            }
+
+    # Render the README template with the contrast info
+    template = env.get_template("README.md")
+    result = template.render(
+        theme=theme_kebab_case,
+        theme_title=theme.replace("_", " ").title(),
+        theme_docstring=getdoc(theme_cls),
+        background_hex=hexstr_without_hash(style.background_color),
+        highlight_hex=hexstr_without_hash(style.highlight_color),
+        colors_hex=foreground_colors.values(),
+    )
+
+    # Save the new README file
+    out = REPO / "a11y_pygments" / theme / "README.md"
+    with open(out, "w") as f:
+        logging.info(f"Updating {out}")
+        f.write(result)
+
+
+# You can either update one theme README at a time or all of them at once
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--theme",
+        type=str,
+        default="",
```

Comment:
I think I would use a special name like `:all:` instead of empty, so that maybe empty could be use for something else  (like show help) later on.

Reply:
I changed it to a positional argument, still not sure I'm following best practice though